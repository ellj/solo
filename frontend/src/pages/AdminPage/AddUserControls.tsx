import React from "react";
import { Select, Input } from "solo-uswds";
import { Warehouse } from "solo-types";

interface Props {
  filter: string;
  onFilterChange: React.ChangeEventHandler<HTMLInputElement>;
  warehouses: Warehouse[];
  onWarehouseChange: React.ChangeEventHandler<HTMLSelectElement>;
}

const AddUserControls: React.FC<Props> = ({
  filter,
  onFilterChange,
  warehouses,
  onWarehouseChange,
}) => {
  return (
    <div className="grid-row flex-row flex-align-center padding-bottom-5">
      <div className="margin-right-3">
        <Input
          value={filter}
          onChange={onFilterChange}
          type="text"
          placeholder="Search Users"
        />
      </div>
      <div>
        <Select onChange={onWarehouseChange}>
          {warehouses.map((warehouse) => (
            <option key={warehouse.id} value={warehouse.id}>
              {warehouse.aac}
            </option>
          ))}
        </Select>
      </div>
    </div>
  );
};

export default AddUserControls;
