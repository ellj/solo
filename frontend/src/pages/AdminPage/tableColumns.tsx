import React from "react";
import { Column } from "react-table";
import { LoadingIcon } from "components";
import { WarehouseUser } from "solo-types";
import { Checkbox, Button } from "solo-uswds";

type CreateColumns = (
  modifyWarehouseUser: (userId: number, data: Partial<WarehouseUser>) => void,
  updateUserPermissons: (userId: number) => void
) => Column<WarehouseUser>[];

const createColumns: CreateColumns = (
  modifyWarehouseUser,
  updateUserPermissons
) => [
  {
    Header: "Loading",
    id: "loading",
    disableSortBy: true,
    Cell: ({
      row: {
        original: { loading, error }
      }
    }) => <LoadingIcon loading={loading} error={error} />
  },
  {
    Header: "AAC",
    accessor: "aac"
  },
  {
    Header: "First Name",
    accessor: "first_name"
  },
  {
    Header: "Last Name",
    accessor: "last_name"
  },
  {
    Header: "DODID",
    accessor: "username"
  },
  {
    Header: "D6T",
    id: "canD6T",
    disableSortBy: true,
    Cell: ({
      row: {
        original: { canD6T, id }
      }
    }) => {
      return (
        <div className="padding-left-2">
          <Checkbox
            data-testid="has-d6t-checkbox"
            checked={canD6T}
            onChange={e => {
              modifyWarehouseUser(id, {
                canD6T: e.target.checked
              });
            }}
          />
        </div>
      );
    }
  },
  {
    Header: "COR",
    id: "canCOR",
    disableSortBy: true,
    Cell: ({
      row: {
        original: { id, canCOR }
      }
    }) => {
      return (
        <div className="padding-left-2">
          <Checkbox
            data-testid="has-cor-checkbox"
            checked={canCOR}
            onChange={e => {
              modifyWarehouseUser(id, {
                canCOR: e.target.checked
              });
            }}
          />
        </div>
      );
    }
  },
  {
    Header: "Submit Permissions",
    id: "submit",
    disableSortBy: true,
    Cell: ({
      row: {
        original: { id, hasModified }
      }
    }) => {
      return (
        <div className="padding-left-2">
          <Button
            disabled={!hasModified}
            onClick={() => {
              updateUserPermissons(id);
            }}
          >
            Submit
          </Button>
        </div>
      );
    }
  }
];

export default createColumns;
