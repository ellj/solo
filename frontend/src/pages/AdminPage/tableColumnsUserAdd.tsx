import React from "react";
import { Column } from "react-table";
import { LoadingIcon } from "components";
import { WarehouseMember } from "solo-types";
import { Checkbox, Button } from "solo-uswds";

type CreateColumns = (
  modifyUser: (userId: number, data: Partial<WarehouseMember>) => void,
  addUserToWarehouse: (userId: number) => void,
  selectedAAC?: string | null
) => Column<WarehouseMember>[];

const createColumns: CreateColumns = (
  modifyUser,
  addUserToWarehouse,
  selectedAAC = ""
) => [
  {
    Header: "Loading",
    id: "loading",
    disableSortBy: true,
    Cell: ({
      row: {
        original: { loading, error },
      },
    }) => <LoadingIcon loading={loading} error={error} />,
  },
  {
    Header: "AAC",
    id: "warehouse",
    accessor: () => selectedAAC,
  },
  {
    Header: "First Name",
    accessor: "user.first_name",
  },
  {
    Header: "Last Name",
    accessor: "user.last_name",
  },
  {
    Header: "DODID",
    accessor: "user.username",
  },
  {
    Header: "D6T",
    id: "d6t_permission",
    disableSortBy: true,
    Cell: ({
      row: {
        original: {
          d6t_permission,
          user: { id },
        },
      },
    }) => {
      return (
        <div className="padding-left-2">
          <Checkbox
            data-testid="has-d6t-checkbox"
            checked={d6t_permission}
            onChange={(e) => {
              modifyUser(id, {
                d6t_permission: e.target.checked,
              });
            }}
          />
        </div>
      );
    },
  },
  {
    Header: "COR",
    id: "cor_permission",
    disableSortBy: true,
    Cell: ({
      row: {
        original: {
          user: { id },
          cor_permission,
        },
      },
    }) => {
      return (
        <div className="padding-left-2">
          <Checkbox
            data-testid="has-cor-checkbox"
            checked={cor_permission}
            onChange={(e) => {
              modifyUser(id, {
                cor_permission: e.target.checked,
              });
            }}
          />
        </div>
      );
    },
  },
  {
    Header: "Submit Permissions",
    id: "submit",
    disableSortBy: true,
    Cell: ({
      row: {
        original: {
          user: { id },
          hasModified,
        },
      },
    }) => {
      return (
        <div className="padding-left-2">
          <Button
            disabled={!hasModified}
            onClick={() => {
              addUserToWarehouse(id);
            }}
          >
            Submit
          </Button>
        </div>
      );
    },
  },
];

export default createColumns;
