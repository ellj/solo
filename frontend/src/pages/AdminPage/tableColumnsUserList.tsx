import React from "react";
import { Column } from "react-table";
import { LoadingIcon } from "components";
import { WarehouseMember } from "solo-types";
import { Checkbox, Button } from "solo-uswds";

type CreateColumns = (
  modifyWarehouseUser: (userId: number, data: Partial<WarehouseMember>) => void,
  updateUserPermissons: (userId: number) => void
) => Column<WarehouseMember>[];

const createColumns: CreateColumns = (
  modifyWarehouseUser,
  updateUserPermissons
) => [
  {
    Header: "Loading",
    id: "loading",
    disableSortBy: true,
    Cell: ({
      row: {
        original: { loading, error },
      },
    }) => <LoadingIcon loading={loading} error={error} />,
  },
  {
    Header: "AAC",
    accessor: "warehouse",
  },
  {
    Header: "First Name",
    accessor: "user.first_name",
  },
  {
    Header: "Last Name",
    accessor: "user.last_name",
  },
  {
    Header: "DODID",
    accessor: "user.username",
  },
  {
    Header: "D6T",
    id: "d6t_permission",
    disableSortBy: true,
    Cell: ({
      row: {
        original: { d6t_permission, id },
      },
    }) => {
      return (
        <div className="padding-left-2">
          <Checkbox
            data-testid="has-d6t-checkbox"
            checked={d6t_permission}
            onChange={(e) => {
              modifyWarehouseUser(id, {
                d6t_permission: e.target.checked,
              });
            }}
          />
        </div>
      );
    },
  },
  {
    Header: "COR",
    id: "cor_permission",
    disableSortBy: true,
    Cell: ({
      row: {
        original: { id, cor_permission },
      },
    }) => {
      return (
        <div className="padding-left-2">
          <Checkbox
            data-testid="has-cor-checkbox"
            checked={cor_permission}
            onChange={(e) => {
              modifyWarehouseUser(id, {
                cor_permission: e.target.checked,
              });
            }}
          />
        </div>
      );
    },
  },
  {
    Header: "Submit Permissions",
    id: "submit",
    disableSortBy: true,
    Cell: ({
      row: {
        original: { id, hasModified },
      },
    }) => {
      return (
        <div className="padding-left-2">
          <Button
            disabled={!hasModified}
            onClick={() => {
              updateUserPermissons(id);
            }}
          >
            Submit
          </Button>
        </div>
      );
    },
  },
];

export default createColumns;
