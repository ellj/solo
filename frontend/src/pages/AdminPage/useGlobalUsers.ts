import { useState, useCallback, useEffect, useRef } from "react";
import { LoadingStatus, User, WarehouseMember, Warehouse } from "solo-types";
import { useAuthContext } from "context";

type Timer = undefined | number | ReturnType<typeof setTimeout>;

const usersToWarehouseUsers = (users: User[]): WarehouseMember[] =>
  users.map((user) => ({
    user,
    id: 0,
    loading: false,
    cor_permission: false,
    d6t_permission: false,
    manager: false,
    warehouse: "",
    hasModified: true,
  }));

const useGlobalUsers = () => {
  const { apiCall } = useAuthContext();
  const [loadingStatus, setLoadingStatus] = useState<LoadingStatus>({
    loading: false,
  });
  const [usersFound, setUsersFound] = useState<WarehouseMember[]>([]);
  const [warehouses, setWarehouses] = useState<Warehouse[]>([]);
  const [selectedWarehouse, setSelectedWarehouse] = useState<Warehouse | null>(
    null
  );
  const [filter, setFilter] = useState<string>("");
  const skipFirstRender = useRef(true);

  const searchUsers = useCallback(
    async (query: string) => {
      if (!selectedWarehouse?.id) {
        return;
      }
      setLoadingStatus({
        loading: true,
        error: false,
      });
      try {
        const url = `/warehouse/${selectedWarehouse.id}/eligibleusers/?q=${query}`;
        const users = await apiCall<User[]>(url, {
          method: "GET",
        });
        setUsersFound(usersToWarehouseUsers(users));
        setLoadingStatus({
          loading: false,
        });
      } catch (e) {
        setLoadingStatus({
          loading: false,
          error: e.message ?? "Error",
        });
      }
    },
    [setLoadingStatus, setUsersFound, apiCall, selectedWarehouse]
  );

  const onFilterChange: React.ChangeEventHandler<HTMLInputElement> = (e) => {
    setFilter(e.currentTarget.value);
  };

  const getMembershipByUserId = useCallback(
    (userId: number) => {
      return usersFound.find(
        (membership) => membership.user.id === userId
      ) as WarehouseMember;
    },
    [usersFound]
  );

  const modifyMembershipByUserId = useCallback(
    (userId: number, data: Partial<WarehouseMember>) => {
      setUsersFound((usersFound) =>
        usersFound.map((membership) =>
          membership.user.id === userId
            ? { ...membership, ...data }
            : membership
        )
      );
    },
    []
  );

  const removeMembershipByUserId = useCallback((userId: number) => {
    setUsersFound((usersFound) =>
      usersFound.filter((membership) => membership.user.id !== userId)
    );
  }, []);

  const addUserToWarehouse = useCallback(
    async (userId: number, warehouseId: number) => {
      modifyMembershipByUserId(userId, {
        loading: true,
        error: false,
      });
      try {
        const { d6t_permission, cor_permission } = getMembershipByUserId(
          userId
        );
        const result = await apiCall<WarehouseMember>(`/warehouseusers/`, {
          method: "POST",
          body: JSON.stringify({
            user_id: userId,
            warehouse_id: warehouseId,
            cor_permission,
            d6t_permission,
          }),
        });
        removeMembershipByUserId(userId);
        return result;
      } catch (e) {
        modifyMembershipByUserId(userId, {
          loading: false,
          error: true,
          message: e.message || "Something went wrong",
        });
      }
    },
    [apiCall, modifyMembershipByUserId, getMembershipByUserId]
  );

  const listManagedWarehouses = useCallback(async () => {
    try {
      const warehouses = await apiCall<Warehouse[]>("/warehouse/", {
        method: "GET",
      });
      setWarehouses(warehouses);
    } catch (e) {
      setWarehouses([
        { id: 1, aac: "test1" },
        { id: 2, aac: "test2" },
      ]);
    }
  }, [setWarehouses]);

  const onSelectWarehouse: React.ChangeEventHandler<HTMLSelectElement> = useCallback(
    (e) => {
      setSelectedWarehouse(
        warehouses.find(
          (warehouse) => warehouse.id === parseInt(e.currentTarget.value)
        ) ?? null
      );
    },
    [warehouses]
  );

  useEffect(() => {
    let timeout: Timer = undefined;
    if (skipFirstRender.current) {
      skipFirstRender.current = false;
    } else if (filter) {
      timeout = setTimeout(() => searchUsers(filter), 500);
    }
    return () => clearTimeout(timeout as number);
  }, [filter]);

  useEffect(() => {
    listManagedWarehouses();
  }, []);

  useEffect(() => {
    setSelectedWarehouse(warehouses[0] ?? null);
  }, [warehouses]);

  useEffect(() => {
    setFilter("");
    setUsersFound([]);
  }, [selectedWarehouse]);

  return {
    loadingStatus,
    searchUsers,
    usersFound,
    filter,
    onFilterChange,
    addUserToWarehouse,
    warehouses,
    selectedWarehouse,
    onSelectWarehouse,
    modifyMembershipByUserId,
    listManagedWarehouses,
  };
};

export default useGlobalUsers;
