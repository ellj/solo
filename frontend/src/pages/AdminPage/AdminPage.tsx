import React, { useMemo, useState, useEffect, useCallback } from "react";
import { Title, Table, SelectFilterControls } from "components";
import { Input, Select } from "solo-uswds";
import createColumns from "./tableColumns";
import useWarehouseUsers from "./useWarehouseUsers";
import useGlobalUsers from "./useGlobalUsers";
import { WarehouseUser } from "solo-types";

const filterable = [
  { name: "Last Name", value: "sdn" },
  { name: "DODID", value: "nsn" }
];

const setGlobalFilter = () => {};

const AdminPage: React.FC = () => {
  const [userFilter, setUserFilter] = useState("");
  const [selectedWarehouseId, setSelectedWarehouseId] = useState<number | null>(
    null
  );
  const {
    //loadingStatus,
    modifyWarehouseUser,
    updateUserPermissons,
    // pageCount,
    users
  } = useWarehouseUsers();
  const {
    usersFound,
    searchUsers,
    modifyUser,
    addUserToWarehouse,
    warehouses
  } = useGlobalUsers();

  const tableColumns = useMemo(
    () => createColumns(modifyWarehouseUser, updateUserPermissons),
    [modifyWarehouseUser, updateUserPermissons]
  );

  const onAddUserToWarehouse = useCallback(
    (userId: number) => {
      if (!selectedWarehouseId) {
        // set some error state here
        console.error("must select a warehouse...");
      } else {
        return addUserToWarehouse(userId, selectedWarehouseId);
      }
    },
    [selectedWarehouseId]
  );

  const foundUsersTableColumns = useMemo(
    () => createColumns(modifyUser, onAddUserToWarehouse),
    [modifyUser, addUserToWarehouse]
  );

  const onUserFilterChange: React.ChangeEventHandler<HTMLInputElement> = e => {
    setUserFilter(e.currentTarget.value);
  };

  useEffect(() => {
    const timeout = setTimeout(() => {
      searchUsers(userFilter);
    }, 300);
    return () => {
      clearTimeout(timeout);
    };
  }, [userFilter]);

  return (
    <div className="tablet:margin-x-8 overflow-x-auto">
      <Title>User Administration</Title>
      <h2>All Users with the same AAC</h2>
      <SelectFilterControls options={filterable} onSubmit={setGlobalFilter} />
      <Table<WarehouseUser> columns={tableColumns} data={users} />
      <h2>Add users</h2>
      <Input
        defaultMargin
        value={userFilter}
        onChange={onUserFilterChange}
        type="text"
        placeholder="Search Users"
      />
      <Select
        onChange={e => setSelectedWarehouseId(parseInt(e.currentTarget.value))}
      >
        {warehouses.map(warehouse => (
          <option key={warehouse.id} value={warehouse.id}>
            {warehouse.aac}
          </option>
        ))}
      </Select>
      {usersFound.length && (
        <Table<WarehouseUser>
          columns={foundUsersTableColumns}
          data={usersFound}
        />
      )}
    </div>
  );
};

export default AdminPage;
