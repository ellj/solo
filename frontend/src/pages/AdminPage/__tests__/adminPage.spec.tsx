import React from "react";
import AdminPage from "../AdminPage";
import { render, fireEvent, wait } from "test-utils";
import {
  defaultUserApiResponse,
  defaultWarehouseMemberApiResponse,
  defaultWareHouseMember,
} from "solo-types";

describe("AdminPage Component", () => {
  const fetchMock = jest.fn();
  const warehouseFetchMock = jest.fn();
  const membersFetchMock = jest.fn();
  const defaultUser = defaultUserApiResponse.results[0];

  beforeEach(() => {
    warehouseFetchMock.mockResolvedValue([
      { id: 1, aac: "testwarehouse1" },
      { id: 2, aac: "testwarehouse2" },
    ]);
    membersFetchMock.mockResolvedValue(defaultWarehouseMemberApiResponse);
    fetchMock.mockImplementation((url: string, ...rest) => {
      if (url === "/warehouse/") {
        return warehouseFetchMock(url, ...rest);
      }
      return membersFetchMock(url, ...rest);
    });
  });

  afterEach(() => {
    warehouseFetchMock.mockReset();
    membersFetchMock.mockReset();
    fetchMock.mockReset();
  });

  const renderAndWaitForData = async () => {
    const { getByText, ...rest } = render(<AdminPage />, {
      authContext: {
        apiCall: fetchMock,
      },
    });
    await wait(() => {
      expect(fetchMock).toHaveBeenCalledTimes(2);
      expect(getByText(defaultUser.username)).toBeInTheDocument();
      expect(getByText(defaultUser.last_name)).toBeInTheDocument();
      expect(getByText(defaultUser.first_name)).toBeInTheDocument();
    });
    return {
      getByText,
      ...rest,
    };
  };

  it("matches snapshot", async () => {
    const { asFragment } = await renderAndWaitForData();
    expect(asFragment()).toMatchSnapshot();
  });

  it("fails to fetch users due to network error", async () => {
    membersFetchMock.mockRejectedValue(new Error("network error"));
    const { queryAllByTestId } = render(<AdminPage />, {
      authContext: {
        apiCall: fetchMock,
      },
    });
    await wait(() => {
      expect(membersFetchMock).toHaveBeenCalled();
      // update checks here once api is integrated
      expect(queryAllByTestId("has-cor-checkbox").length).toBeGreaterThan(0);
    });
  });

  it("click on checkboxes and change permissions", async () => {
    membersFetchMock.mockResolvedValue({
      ...defaultWarehouseMemberApiResponse,
      results: [
        {
          ...defaultWareHouseMember,
          d6t_permission: false,
          cor_permission: false,
        },
      ],
    });
    const { getByTestId, getByText, container } = await renderAndWaitForData();
    await wait(() => {
      expect(getByTestId("has-cor-checkbox")).not.toBeChecked();
      expect(getByTestId("has-d6t-checkbox")).not.toBeChecked();
      expect(getByText("Submit")).toBeDisabled();
    });
    fireEvent.click(getByTestId("has-cor-checkbox"));
    fireEvent.click(getByTestId("has-d6t-checkbox"));
    await wait(() => {
      expect(getByTestId("has-cor-checkbox")).toBeChecked();
      expect(getByTestId("has-d6t-checkbox")).toBeChecked();
      expect(getByText("Submit")).not.toBeDisabled();
    });
    membersFetchMock.mockResolvedValue({});
    fireEvent.click(getByText("Submit"));
    await wait(() => {
      expect(membersFetchMock).toHaveBeenCalledTimes(2);
      expect(membersFetchMock.mock.calls[1][0]).toEqual(
        `/warehouseusers/${defaultWareHouseMember.id}/`
      );
      expect(membersFetchMock.mock.calls[1][1]).toMatchObject({
        method: "PATCH",
        body: JSON.stringify({
          d6t_permission: true,
          cor_permission: true,
        }),
      });
      expect(container.querySelector("svg.fa-check")).toBeInTheDocument();
    });
  });

  it("submit user permissions network error shows error indicator icon", async () => {
    membersFetchMock.mockResolvedValue({
      ...defaultWarehouseMemberApiResponse,
      results: [
        {
          ...defaultWareHouseMember,
          d6t_permission: false,
          cor_permission: false,
        },
      ],
    });
    const { getByTestId, getByText, container } = await renderAndWaitForData();
    await wait(() => {
      expect(getByTestId("has-cor-checkbox")).not.toBeChecked();
      expect(getByTestId("has-d6t-checkbox")).not.toBeChecked();
      expect(getByText("Submit")).toBeDisabled();
    });
    fireEvent.click(getByTestId("has-cor-checkbox"));
    fireEvent.click(getByTestId("has-d6t-checkbox"));
    await wait(() => {
      expect(getByTestId("has-cor-checkbox")).toBeChecked();
      expect(getByTestId("has-d6t-checkbox")).toBeChecked();
      expect(getByText("Submit")).not.toBeDisabled();
    });
    membersFetchMock.mockRejectedValue(new Error());
    fireEvent.click(getByText("Submit"));
    await wait(() => {
      expect(membersFetchMock).toHaveBeenCalledTimes(2);
      expect(
        container.querySelector("svg.fa-exclamation-circle")
      ).toBeInTheDocument();
    });
  });

  it("submit user permissions network error with message shows error indicator icon", async () => {
    membersFetchMock.mockResolvedValue({
      ...defaultUserApiResponse,
      results: [
        {
          ...defaultWareHouseMember,
          d6t_permission: false,
          cor_permission: false,
        },
      ],
    });
    const { getByTestId, getByText, container } = await renderAndWaitForData();
    await wait(() => {
      expect(getByTestId("has-cor-checkbox")).not.toBeChecked();
      expect(getByTestId("has-d6t-checkbox")).not.toBeChecked();
      expect(getByText("Submit")).toBeDisabled();
    });
    fireEvent.click(getByTestId("has-cor-checkbox"));
    fireEvent.click(getByTestId("has-d6t-checkbox"));
    await wait(() => {
      expect(getByTestId("has-cor-checkbox")).toBeChecked();
      expect(getByTestId("has-d6t-checkbox")).toBeChecked();
      expect(getByText("Submit")).not.toBeDisabled();
    });
    const err = new Error();
    err.message = "some error message";
    membersFetchMock.mockRejectedValue(err);
    fireEvent.click(getByText("Submit"));
    await wait(() => {
      expect(membersFetchMock).toHaveBeenCalledTimes(2);
      expect(
        container.querySelector("svg.fa-exclamation-circle")
      ).toBeInTheDocument();
    });
  });

  it("checking permission options only updates user for that row", async () => {
    membersFetchMock.mockResolvedValue({
      ...defaultWarehouseMemberApiResponse,
      results: [
        {
          ...defaultWareHouseMember,
          id: 5,
        },
        {
          ...defaultWareHouseMember,
          id: 42,
          user: {
            ...defaultUser,
            first_name: "Jeff",
            last_name: "Hackshaw",
            username: "5678567567",
          },
        },
      ],
    });
    const { getAllByTestId, getAllByText } = await renderAndWaitForData();
    await wait(() => {
      expect(getAllByTestId("has-cor-checkbox")).toHaveLength(2);
      expect(getAllByTestId("has-cor-checkbox")[0]).not.toBeChecked();
      expect(getAllByTestId("has-cor-checkbox")[1]).not.toBeChecked();
      expect(getAllByText("Submit")).toHaveLength(2);
      expect(getAllByText("Submit")[0]).toBeDisabled();
      expect(getAllByText("Submit")[1]).toBeDisabled();
    });
    fireEvent.click(getAllByTestId("has-cor-checkbox")[0]);
    await wait(() => {
      expect(getAllByTestId("has-cor-checkbox")[0]).toBeChecked();
      expect(getAllByText("Submit")[0]).not.toBeDisabled();
    });
  });
});
