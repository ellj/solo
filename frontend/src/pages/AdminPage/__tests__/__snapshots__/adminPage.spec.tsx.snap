// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminPage Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="tablet:margin-x-8 overflow-x-auto overflow-y-hidden"
  >
    <h1
      class="usa-prose"
    >
      User Administration
    </h1>
    <form
      class="grid-row flex-row flex-justify-center flex-align-center"
    >
      <div
        class="grid-col-full tablet:grid-col-2 margin-x-2"
      >
        <select
          class="usa-select"
          placeholder="Field"
        >
          <option
            value="lastName"
          >
            Last Name
          </option>
          <option
            value="dodid"
          >
            DODID
          </option>
        </select>
      </div>
      <div
        class="grid-col-10 tablet:grid-col-3"
      >
        <input
          class="usa-input noOutline"
          placeholder="Search"
          type="text"
          value=""
        />
      </div>
      <div
        class="grid-col-auto"
      >
        <button
          class="usa-button margin-top-1 square"
          type="submit"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-search fa-w-16 "
            data-icon="search"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
              fill="currentColor"
            />
          </svg>
        </button>
      </div>
    </form>
    <table
      class="usa-table grid-col-12 usa-table--borderless tableBorders"
      role="table"
    >
      <thead>
        <tr
          role="row"
        >
          <th
            colspan="1"
            role="columnheader"
          >
            <div
              class="display-flex flex-justify flex-align-center flex-no-wrap text-no-wrap"
            >
              Loading
            </div>
          </th>
          <th
            colspan="1"
            role="columnheader"
            style="cursor: pointer;"
            title="Toggle SortBy"
          >
            <div
              class="display-flex flex-justify flex-align-center flex-no-wrap text-no-wrap"
            >
              AAC
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-chevron-up fa-w-14 margin-left-05"
                data-icon="chevron-up"
                data-prefix="fas"
                focusable="false"
                role="img"
                style="visibility: hidden;"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M240.971 130.524l194.343 194.343c9.373 9.373 9.373 24.569 0 33.941l-22.667 22.667c-9.357 9.357-24.522 9.375-33.901.04L224 227.495 69.255 381.516c-9.379 9.335-24.544 9.317-33.901-.04l-22.667-22.667c-9.373-9.373-9.373-24.569 0-33.941L207.03 130.525c9.372-9.373 24.568-9.373 33.941-.001z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </th>
          <th
            colspan="1"
            role="columnheader"
            style="cursor: pointer;"
            title="Toggle SortBy"
          >
            <div
              class="display-flex flex-justify flex-align-center flex-no-wrap text-no-wrap"
            >
              First Name
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-chevron-up fa-w-14 margin-left-05"
                data-icon="chevron-up"
                data-prefix="fas"
                focusable="false"
                role="img"
                style="visibility: hidden;"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M240.971 130.524l194.343 194.343c9.373 9.373 9.373 24.569 0 33.941l-22.667 22.667c-9.357 9.357-24.522 9.375-33.901.04L224 227.495 69.255 381.516c-9.379 9.335-24.544 9.317-33.901-.04l-22.667-22.667c-9.373-9.373-9.373-24.569 0-33.941L207.03 130.525c9.372-9.373 24.568-9.373 33.941-.001z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </th>
          <th
            colspan="1"
            role="columnheader"
            style="cursor: pointer;"
            title="Toggle SortBy"
          >
            <div
              class="display-flex flex-justify flex-align-center flex-no-wrap text-no-wrap"
            >
              Last Name
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-chevron-up fa-w-14 margin-left-05"
                data-icon="chevron-up"
                data-prefix="fas"
                focusable="false"
                role="img"
                style="visibility: hidden;"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M240.971 130.524l194.343 194.343c9.373 9.373 9.373 24.569 0 33.941l-22.667 22.667c-9.357 9.357-24.522 9.375-33.901.04L224 227.495 69.255 381.516c-9.379 9.335-24.544 9.317-33.901-.04l-22.667-22.667c-9.373-9.373-9.373-24.569 0-33.941L207.03 130.525c9.372-9.373 24.568-9.373 33.941-.001z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </th>
          <th
            colspan="1"
            role="columnheader"
            style="cursor: pointer;"
            title="Toggle SortBy"
          >
            <div
              class="display-flex flex-justify flex-align-center flex-no-wrap text-no-wrap"
            >
              DODID
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-chevron-up fa-w-14 margin-left-05"
                data-icon="chevron-up"
                data-prefix="fas"
                focusable="false"
                role="img"
                style="visibility: hidden;"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M240.971 130.524l194.343 194.343c9.373 9.373 9.373 24.569 0 33.941l-22.667 22.667c-9.357 9.357-24.522 9.375-33.901.04L224 227.495 69.255 381.516c-9.379 9.335-24.544 9.317-33.901-.04l-22.667-22.667c-9.373-9.373-9.373-24.569 0-33.941L207.03 130.525c9.372-9.373 24.568-9.373 33.941-.001z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </th>
          <th
            colspan="1"
            role="columnheader"
          >
            <div
              class="display-flex flex-justify flex-align-center flex-no-wrap text-no-wrap"
            >
              D6T
            </div>
          </th>
          <th
            colspan="1"
            role="columnheader"
          >
            <div
              class="display-flex flex-justify flex-align-center flex-no-wrap text-no-wrap"
            >
              COR
            </div>
          </th>
          <th
            colspan="1"
            role="columnheader"
          >
            <div
              class="display-flex flex-justify flex-align-center flex-no-wrap text-no-wrap"
            >
              Submit Permissions
            </div>
          </th>
        </tr>
      </thead>
      <tbody
        role="rowgroup"
      >
        <tr
          role="row"
        >
          <td
            class="td"
            role="cell"
          >
            <span>
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-check fa-w-16 "
                data-icon="check"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
                  fill="currentColor"
                />
              </svg>
            </span>
          </td>
          <td
            class="td"
            role="cell"
          >
            testaac
          </td>
          <td
            class="td"
            role="cell"
          >
            testFirst
          </td>
          <td
            class="td"
            role="cell"
          >
            testLast
          </td>
          <td
            class="td"
            role="cell"
          >
            1234123123
          </td>
          <td
            class="td"
            role="cell"
          >
            <div
              class="padding-left-2"
            >
              <input
                class="usa-checkbox__input inp"
                data-testid="has-d6t-checkbox"
                type="checkbox"
              />
            </div>
          </td>
          <td
            class="td"
            role="cell"
          >
            <div
              class="padding-left-2"
            >
              <input
                class="usa-checkbox__input inp"
                data-testid="has-cor-checkbox"
                type="checkbox"
              />
            </div>
          </td>
          <td
            class="td"
            role="cell"
          >
            <div
              class="padding-left-2"
            >
              <button
                class="usa-button"
                disabled=""
              >
                Submit
              </button>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
    <h2>
      Add users
    </h2>
    <div
      class="grid-row flex-row flex-align-center padding-bottom-5"
    >
      <div
        class="margin-right-3"
      >
        <input
          class="usa-input noMargin noOutline"
          placeholder="Search Users"
          type="text"
          value=""
        />
      </div>
      <div>
        <select
          class="usa-select noMargin noOutline"
        >
          <option
            value="1"
          >
            testwarehouse1
          </option>
          <option
            value="2"
          >
            testwarehouse2
          </option>
        </select>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
